spec/promoted/ruby/client_spec.rb

```diff
@@ -1,9 +1,53 @@
+require 'spec_helper'
+
 RSpec.describe Promoted::Ruby::Client do
+  let!(:input) { SAMPLE_INPUT }
+  let!(:logging_json) { Promoted::Ruby::Client.prepare_for_logging(input) }
+
   it "has a version number" do
     expect(Promoted::Ruby::Client::VERSION).not_to be nil
   end
 
-  it "does something useful" do
-    expect(false).to eq(true)
+  context "prepare_for_logging when no limit is set" do
+    it "has user_info set" do
+      expect(logging_json[:user_info]).not_to be nil
+      expect(logging_json[:user_info][:user_id]).to eq(input.dig('request', 'user_info', 'user_id'))
+      expect(logging_json[:user_info][:log_user_id]).to eq(input.dig('request', 'user_info', 'log_user_id'))
+    end
+
+    it "should not have full_insertion" do
+      expect(logging_json[:full_insertion]).to be nil
```

Comment:
In a follow up PR, please add more detailed asserts.  This won't catch smaller, critical bugs.